src/components/UserSelector.tsx

```diff
@@ -1,19 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+interface Props {
+  users: User[];
+  selectedUser: User | undefined;
+  setSelectedUser: React.Dispatch<React.SetStateAction<User | undefined>>;
+  setSelectedPost: React.Dispatch<React.SetStateAction<Post | undefined>>;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selectedUser,
+  setSelectedUser = () => {},
+  setSelectedPost = () => {},
+}) => {
+  const [isActiveDropDown, setIsActiveDropDown] = useState<boolean>(false);
+
+  const changeDropDownState = () => {
+    setIsActiveDropDown(!isActiveDropDown);
+  };
+
+  const closeDropDown = () => {
+    setTimeout(() => {
+      setIsActiveDropDown(false);
+    }, 150); // Maybe more
```

Comment:
Google best [practice](https://www.joshwcomeau.com/animation/css-transitions/)
And remove comment
Take responsibility to your code 😃